cobalt-ast/src/types.rs

```diff
@@ -766,6 +768,43 @@ impl<'ctx> NominalInfo<'ctx> {
         let mut c = 0u8;
         buf.read_exact(std::slice::from_mut(&mut c))?;
         let no_auto_drop = c != 0;
-        Ok(Self { dtor, no_auto_drop })
+        Ok(Self {
+            dtor,
+            no_auto_drop,
+            is_linear_type: false,
```

Comment:
This needs proper serialization/deserialization. You could store it as a bit in that byte (make sure to account for multiple bits being set in `no_auto_drop`).